api/LMPlatform.UI/Controllers/AccountController.cs

```diff
@@ -310,6 +317,57 @@ public string GenerateBase64(HttpPostedFileBase file)
             return "data:image/png;base64," + Convert.ToBase64String(thePictureAsBytes);
         }
 
+        [JwtAuth]
+        [HttpPost]
+        public ActionResult DeleteAccount()
+        {
+            var model = new PersonalDataViewModel();
+            var id = UsersManagementService.GetUser(model.UserName).Id;
+
+            var deleted = false;
+
+            try
+            {
+                if (Roles.IsUserInRole("lector"))
+                {
+                    var lecturer = this.LecturerManagementService.GetLecturer(id);
+
+                    if (lecturer is null)
+                    {
+                        return StatusCode(HttpStatusCode.BadRequest);
+                    }
+
+                    if (!(lecturer.SubjectLecturers is null) && lecturer.SubjectLecturers.Any() &&
+                        lecturer.SubjectLecturers.All(e => e.Subject.IsArchive))
+                    {
+                        foreach (var lecturerSubjectLecturer in lecturer.SubjectLecturers)
+                        {
+                            this.LecturerManagementService.DisjoinOwnerLector(lecturerSubjectLecturer.SubjectId, id);
+                        }
+                    }
+
+                    deleted = this.LecturerManagementService.DeleteLecturer(id);
+                    this.ElasticManagementService.DeleteLecturer(id);
+                }
+                else
+                {
+                    if (StudentManagementService.GetStudent(id) is null)
```

Comment:
we don;t need to check that student exist. We have the same check into `DeleteStudent`